packages/babel-plugin-proposal-explicit-resource-management/test/fixtures/integration/async-to-generator/output.js

```diff
@@ -1,8 +1,5 @@
 function fn() {
-  return _fn.apply(this, arguments);
-}
-function _fn() {
-  _fn = babelHelpers.asyncToGenerator(function* () {
+  return babelHelpers.callAsync(function* () {
```

Comment:
Was the intention of the old code to keep this as a top-level function, so we don't need to reevaluate it constantly? I don't think it achieved that, but it would help overall performance.

Reply:
Initially it was a variable declaration + a function declaration, but later it became two function declarations to fix the hoisting problem.
Perhaps initially creating a new closure might be to facilitate working with `bluebird`/`regenerator`.
Because they're better off calling `regenerator.mark`/`bluebird.coroutine` only once on the function body.
(I'm not sure about this, just guessing)